orangecontrib/prototypes/widgets/owunsdata.py

```diff
@@ -0,0 +1,383 @@
+
+        nmbox = gui.indentedBox(self.radiobox)
+
+        gui.spin(
+            nmbox, self, "last_measurements", 1, 10000000, label="Last measurements",
+            controlWidth=60, callback=self.number_changed)
+
+        self.radiobox.setDisabled(not self.query_controls)
+
+
+        self.objects = {}
+        self.lb_objects = gui.listBox(self.controlArea, self, "selected", "lb_labels", box="Deployments",
+                                    callback=self.sel_changed,
+                                      sizeHint=QSize(300, 300))
+        self.lb_objects.setFocusPolicy(Qt.NoFocus)
+
+        out_box = gui.vBox(self.mainArea, "Output")
+
+        self.out_box = gui.radioButtons(out_box, self, "output_type", callback=self.output_changed)
+
+        gui.appendRadioButton(self.out_box, "Matrix")
+        gui.appendRadioButton(self.out_box, "Tensor")
+
+        self.apply_button = gui.button(self.mainArea, self, self.tr("&Apply"),
+                                       callback=self.apply)
+
+        self.out_box.setDisabled(not self.query_controls)
+
+
+        self._executor = ThreadPoolExecutor(max_workers=1)
+        self._executor.submit(self.get_deployments())
+
+
+
+    def get_deployments(self):
+        url = self.BASE_URL + 'deployment'
+        response = requests.get(url, headers={'content-type': 'application/json', 'accept': 'application/json'})
+        if response.status_code == 200:
+            deployments = response.json()
+
+            for dep in deployments:
+                jsn_data = dep
+                item = QListWidgetItem(dep["name"])
+                item.setData(Qt.UserRole, jsn_data)
+                self.lb_objects.addItem(item)
+
+
+    def radio_changed(self):
+        self.ow_cache = 0
+        print(self.query_type)
+
+    def output_changed(self):
+        self.ow_cache = 0
+        print(self.output_type)
+
+
+    def apply(self):
+        if self.lb_objects.count() != 0:
+            print("getting data")
+            self.Information.getting_data()
+            self.raise_info_getting_data()
+            print("getting data info done")
+            selected_deployment = self.lb_objects.currentItem().data(Qt.UserRole)
+            dep_id = selected_deployment["_id"]
+            url = self.BASE_URL
+
+            if self.query_type == 0:
+                url += 'data/deployment/'
+                url += str(dep_id)
+
+            if self.query_type == 1:
+                url += 'data/deployment/interesting/'
+                url += str(dep_id)
+                url += '/'
+                url += str(self.interval_len)
+                url += '/'
+                url += str(self.n_intervals)
+
+            if self.query_type == 2:
+                print("yeet")
+
+            if self.query_type == 3:
+                url += 'data/deployment/'
+                url += str(dep_id)
+                url += '/seconds/'
+                url += str(self.last_seconds)
+
+            if self.query_type == 4:
```

Comment:
How will you know in 4 months what query type [4](https://en.wikipedia.org/wiki/Magic_number_(programming)) is?